rust/lance/src/io/commit.rs

```diff
@@ -76,11 +76,58 @@ fn manifest_path(base: &Path, version: u64) -> Path {
         .child(format!("{version}.{MANIFEST_EXTENSION}"))
 }
 
-/// Get the latest manifest path
 fn latest_manifest_path(base: &Path) -> Path {
     base.child(LATEST_MANIFEST_NAME)
 }
 
+/// Get the latest manifest path
+async fn current_manifest_path(object_store: &ObjectStore, base: &Path) -> Result<Path> {
+    // TODO: list gives us the size, so we could also return the size of the manifest.
+    // That avoids a HEAD request later.
+    let mut manifest_files = object_store
+        .inner
+        .list(Some(&base.child(VERSIONS_DIR)))
```

Comment:
this might take a long time if they dataset has a lot of versions.

Maybe we could try something like this?
1. read `_latest.manifest`
2. use the version in `_latest.manifest` as the start
3. if we hit a version like `xxxx9999`, list from `10000000` and keep going until we find the latests manifest.

Reply:
nvm, i realized reading `_latest.manifest` is a no starter

Reply:
And we can't list from because we aren't zero-padding.

But also see my comment about performance in description.